### ISSUE TITLE
skill-authoring.md: the prescribed loss-check misfires on every line starting with `-`

### ISSUE THREAD

**Issue by jazorro11** [rebelytics/one-skill-to-rule-them-all#24]
`references/skill-authoring.md` → *Verifying relocations and restructures*, step 2, prescribes:

> 2. Exact-match each non-empty line against the restructured file set (`grep -F`).

Implemented literally — `grep -qF "$line" "$file"` — every line beginning with `-` is parsed as an
option bundle rather than a pattern. `grep` errors out and the line is scored as **missing**.

Since skills are mostly markdown bullet lists, the check cries wolf on the majority of the content
it is meant to protect.

### Reproduction

Two **identical** files:

```bash
printf '# Title\n- a bullet line\nplain line\n' > old.md
cp old.md new.md

# as prescribed:
grep -vE '^\s*$' old.md | while IFS= read -r l; do
  grep -qF "$l" new.md || echo "REPORTED AS LOST: $l"
done
```

Output (GNU grep 3.0):

```
grep: unknown option --
Usage: grep [OPTION]... PATTERN [FILE]...
REPORTED AS LOST: - a bullet line
```

Adding the option terminator fixes it — same command, `grep -qF -- "$l"` — and prints nothing, which
is the correct result for two identical files.

### Why it matters here

I hit this while restructuring a skill under the procedure in that section. The run produced **31
false "missing" lines alongside 35 genuine ones**, and the two were indistinguishable without a
second pass. A verification step is trusted precisely to rule out "did I lose something?", so a
false negative there is worse than no check: it either hides real losses in noise, or burns a review
cycle chasing losses that never happened.

It is also a textbook case of the skill's own **Pre-Flight Principle** — "an embedded command runs
verbatim, unattended, forever, and a subtly wrong command can read as correct on every re-read". The
snippet reads fine; it only fails when executed against realistic input.

### Suggested fix

Give the check as a runnable line with the terminator included, plus a one-clause note on why it is
required:

```bash
grep -qF -- "$line" "$file"   # -- is required: markdown lines starting with `-` are otherwise parsed as options
```

It may be worth sweeping the other reference files for embedded snippets that interpolate file
content into an argument position, since the same failure mode applies wherever that happens.

### Related

Sibling of #16, but distinct and worth fixing together:

- #16 is a **portability** problem — `grep -oP` and bare `sed -i` are GNU-only, so those blocks fail
  on macOS/BSD. It is about *where* the snippet runs.
- This one is an **option-parsing** problem: it fails on every platform, GNU included, whenever the
  *data* happens to start with `-`. It is about *what the snippet is fed*.

The shared root cause is that embedded snippets interpolate untrusted content into argument
position, so a single sweep over all reference files would close both. Happy to send a PR for this
one if that is useful.

### Environment

GNU grep 3.0 (Git Bash on Windows). The behaviour is not platform-specific — it follows from POSIX
option parsing, so it applies to any `grep` unless the pattern is guarded.

*Filed from an install's skill-observation log (Observation 6), surfaced while restructuring an
unrelated skill under the procedure in that section.*

**Comment by rebelytics** [comment]
Thanks, @jazorro11. I will fix this with a PR for v3 where I will add you as a co-author, along with #16, as mentioned by you.

**Comment by rebelytics** [comment]
Fixed in `release/v3.0.0`: the relocation check reads `grep -qF -- "$line" "$file"`, with the reason stated inline, and the rule is generalised to every embedded snippet that interpolates file content into an argument position. The pre-flight principle gained a line on running the literal string from a clean shell for the same reason.

You're credited as co-author on the commit. FYI: I'll be running v3 for a week or so before it goes into main.

**Comment by rebelytics** [comment]
Shipped in v3.0.0, released today. Thanks again for your contribution — the release notes credit everyone whose reports and PRs shaped this version.
